src/FormElement/ScheduleElement.php

```diff
@@ -0,0 +1,628 @@
+                    'id'        => $this->protectId('schedule-recurrences'),
+                    'label'     => $this->translate('Next occurrences'),
+                    'valid'     => function (): bool {
+                        return $this->isValid();
+                    },
+                    'frequency' => function (): Frequency {
+                        if ($this->getFrequency() === static::CUSTOM_EXPR) {
+                            list($rule, $start, $end) = array_values($this->getValue());
+                            $rule = new RRule($rule);
+                        } else {
+                            $rule = RRule::fromFrequency($this->getFrequency());
+                            $start = $this->getPopulatedValue('start', new DateTime());
+                            if (! $start instanceof DateTime) {
+                                $start = new DateTime($start);
+                            }
+
+                            $end = null;
+                            if ($this->getPopulatedValue('use-end-time') === 'y') {
+                                $end = $this->getPopulatedValue('end', new DateTime());
+                                if (! $end instanceof DateTime) {
+                                    $end = new DateTime($end);
+                                }
+                            }
+                        }
+
+                        $rule->startAt($start);
+                        if ($end instanceof DateTime) {
+                            $rule->endAt($end);
+                        }
+
+                        return $rule;
+                    }
+                ])
+            );
+        }
+    }
+
+    /**
+     * Assemble common parts for all the frequencies
+     */
+    private function assembleCommonElements(): void
+    {
+        $repeat = $this->getCustomFrequency();
+        if ($repeat === RRule::WEEKLY) {
+            $text = $this->translate('week(s) on');
+            $max = 53;
+        } elseif ($repeat === RRule::MONTHLY) {
+            $text = $this->translate('month(s)');
+            $max = 12;
+        } else {
+            $text = $this->translate('day(s)');
+            $max = 31;
+        }
+
+        $options = ['min' => 1, 'max' => $max];
+        $this->addElement('number', 'interval', [
+            'class'      => 'autosubmit',
+            'value'      => 1,
+            'min'        => 1,
+            'max'        => $max,
+            'validators' => [new BetweenValidator($options)]
+        ]);
+
+        $numberSpecifier = HtmlElement::create('div', ['class' => 'number-specifier']);
+        $element = $this->getElement('interval');
+        $element->prependWrapper($numberSpecifier);
+
+        $numberSpecifier->prependHtml(HtmlElement::create('span', null, $this->translate('Every')));
+        $numberSpecifier->addHtml($element);
+        $numberSpecifier->addHtml(HtmlElement::create('span', null, $text));
```

Comment:
Please use a `TemplateString` here. Get back to me if you're unsure what to do with it :wink: